### PR TITLE
Completed Trees-1 - Please Review

### PR DESCRIPTION
**Pull request by DhruvJawalkar**

### CODE REVIEW

File: TreeFromInorderAndPreorder.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// Time Complexity : O(n), we would process and build one root node at a time I guess
```
Comment (DhruvJawalkar) [inline TreeFromInorderAndPreorder.java:1]: I wasn't very clear about how to calculate the TC using the master theorem here

File: TreeFromInorderAndPreorder.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
+// I wasn't very clear about how to calculate the TC using the master theorem here
+// Time Complexity : O(n), we would process and build one root node at a time 
+// Space Complexity : O(n) + O(log n) for (preorder, inorder, hashmap) + for call stack 
+// Did this code successfully run on Leetcode : yes
+// Any problem you faced while coding this :
+// This was hard, I did not get the intuition to solve the problem, I was trying to make the tree by traversing the array in reverse given right would be the rightmost element
```
Comment (kannan-mrugapphan) [inline TreeFromInorderAndPreorder.java:6]: The other approach where you have a separate list at each call will build more intuition. I would recommend solving that as well.

**Review by kannan-mrugapphan — COMMENTED** [review]
Nice Work. good Explanation

**Comment by kannan-mrugapphan** [comment]
Nice Work. good Explanation

**Pull request closed without merging**